android/src/main/kotlin/com/appcues/flutter/FlutterElementTargetingStrategy.kt

```diff
@@ -0,0 +1,158 @@
+        return plugin.activity?.getParentView()?.let {
+            val actualPosition = Rect()
+            it.getGlobalVisibleRect(actualPosition)
+
+            val displayMetrics = it.context.resources.displayMetrics
+            val density = displayMetrics.density
+
+            return ViewElement(
+                x = actualPosition.left.toDp(density),
+                y = actualPosition.top.toDp(density),
+                width = actualPosition.width().toDp(density),
+                height = actualPosition.height().toDp(density),
+                selector = null,
+                displayName = null,
+                type = this.javaClass.name,
+                children = widgets.values.toList(),
+            )
+        }
+    }
+
+    override fun inflateSelectorFrom(properties: Map<String, String>): ElementSelector? {
+        return FlutterElementSelector(
+            identifier = properties["identifier"],
+        ).let { if (it.isValid) it else null }
+    }
+
+    override fun captureScreenshot(): Screenshot? {
+        return plugin.activity?.let { activity ->
+            val view = activity.getParentView()
+            val density = view.resources.displayMetrics.density
+
+            // get the root level window insets, which will be used to apply a height correction
+            // to the screenshot below
+            val insets = ViewCompat.getRootWindowInsets(view)?.getInsets(WindowInsetsCompat.Type.systemBars())
+                ?: Insets.NONE
+
+            // This bitmap from the FlutterRenderer contains the top system bar but not the bottom
+            val bitmap = plugin.renderer.bitmap
+
+            // so bottom is subtracted from the screenshot height
+            val width = view.width.toDp(density)
+            val height = (view.height - insets.bottom).toDp(density)
+
+            Screenshot(
+                bitmap = bitmap,
+                size = Size(width, height),
+                // zero out the bottom insets for the screen capture builder usage
+                insets = Insets.of(insets.left, insets.top, insets.right, 0)
+            )
+        }
+
+    }
+
+    fun targetElement(call: MethodCall) {
+        val identifier = call.argument<String>("identifier")
+        val x = call.argument<Double>("x")
+        val y = call.argument<Double>("y")
+        val width = call.argument<Double>("width")
+        val height = call.argument<Double>("height")
+        val type = call.argument<String>("type")
+
+        if (identifier != null && x != null && y != null && width != null && height != null && type != null) {
+            widgets[identifier] = ViewElement(
+                x = x.toInt(),
+                y = y.toInt(),
+                width = width.toInt(),
+                height = height.toInt(),
+                type = type,
+                selector = FlutterElementSelector(identifier),
+                children = null,
+                displayName = identifier
+            )
+        }
+    }
```

Comment:
here is where layout info comes in from the Flutter side, and we capture in our `widgets` collection for later usage